service.js

```diff
@@ -1,57 +1,74 @@
 let _ = require('lodash/fp')
 
-module.exports = (models, modelName) => ({
-  createNested: async record => {
+module.exports = (models, modelName) => {
+  let destroy = _.curry(async (options, record) => {
+    let {soft = false, cascade = false, customDelete, beforeDelete} = options
     let model = models[modelName]
+    let id = record.id
 
-    let associationIds = _.flow(
-      _.map('alias'),
-      _.filter(id => _.isPlainObject(record[id]))
-    )(model.associations)
-
-    let {id} = await model.create(_.omit(associationIds, record)).meta({fetch: true}).then()
+    if (_.isFunction(customDelete)) await customDelete(options, record, model, models)
+    else {
+      // Current cascade implementation supports deleting a matching association as part of the passed in deleted entity (record)
+      // TODO: Allow some sort of automatic cascading delete based on the associations of the deleted entity
+      if (cascade) {
+        await Promise.all(_.map(async association => {
```

Comment:
Is this nested? or just the first level of associations?

Reply:
yes, it is only for first level associations, which is how desroyNested worked before as well, there is no traversing at this point of associations